packages/vaadin-combo-box/src/vaadin-combo-box.d.ts

```diff
@@ -155,94 +158,35 @@ import { ComboBoxDefaultItem, ComboBoxEventMap } from './interfaces';
  * @fires {CustomEvent} selected-item-changed - Fired when the `selectedItem` property changes.
  * @fires {CustomEvent} value-changed - Fired when the `value` property changes.
  */
-declare class ComboBoxElement<TItem = ComboBoxDefaultItem> extends HTMLElement {
-  /**
-   * Focusable element used by vaadin-control-state-mixin
-   */
-  readonly focusElement: HTMLElement;
-
-  autofocus: boolean;
-
-  /**
-   * Set to true to disable this input.
-   */
-  disabled: boolean;
-
-  /**
-   * Set to true to prevent the user from picking a value or typing in the input.
-   */
-  readonly: boolean;
-
-  readonly inputElement: TextFieldElement | undefined;
-
-  /**
-   * The label for this element.
-   */
-  label: string | null | undefined;
-
-  /**
-   * Set to true to mark the input as required.
-   */
-  required: boolean;
-
-  /**
-   * Set to true to prevent the user from entering invalid input.
-   * @attr {boolean} prevent-invalid-input
-   */
-  preventInvalidInput: boolean | null | undefined;
-
-  /**
-   * A pattern to validate the `input` with.
-   */
-  pattern: string | null | undefined;
-
-  /**
-   * The error message to display when the input is invalid.
-   * @attr {string} error-message
-   */
-  errorMessage: string | null | undefined;
-
-  /**
-   * A placeholder string in addition to the label.
-   */
-  placeholder: string;
-
-  /**
-   * String used for the helper text.
-   * @attr {string} helper-text
-   */
-  helperText: string | null | undefined;
-
-  /**
-   * Set to true to display the clear icon which clears the input.
-   * @attr {boolean} clear-button-visible
-   */
-  clearButtonVisible: boolean;
-
+declare class ComboBox<TItem = ComboBoxDefaultItem> extends HTMLElement {
   addEventListener<K extends keyof ComboBoxEventMap<TItem>>(
     type: K,
-    listener: (this: ComboBoxElement<TItem>, ev: ComboBoxEventMap<TItem>[K]) => void,
+    listener: (this: ComboBox<TItem>, ev: ComboBoxEventMap<TItem>[K]) => void,
     options?: boolean | AddEventListenerOptions
   ): void;
 
   removeEventListener<K extends keyof ComboBoxEventMap<TItem>>(
     type: K,
-    listener: (this: ComboBoxElement<TItem>, ev: ComboBoxEventMap<TItem>[K]) => void,
+    listener: (this: ComboBox<TItem>, ev: ComboBoxEventMap<TItem>[K]) => void,
     options?: boolean | EventListenerOptions
   ): void;
 }
 
-interface ComboBoxElement<TItem = ComboBoxDefaultItem>
+interface ComboBox<TItem = ComboBoxDefaultItem>
   extends ElementMixin,
-    ControlStateMixin,
```

Comment:
Move `ComboBoxMixin` at the top to have its property descriptions override those coming from the mixins:

Before:

<img width="415" alt="Screenshot 2021-09-15 at 17 39 01" src="https://user-images.githubusercontent.com/1222264/133454437-542a78ce-a999-417d-9356-9c9fe0c83f4e.png">

After:

<img width="574" alt="Screenshot 2021-09-15 at 17 37 05" src="https://user-images.githubusercontent.com/1222264/133454153-1ab49056-e932-4795-9d21-7ea55c280419.png">

Reply:
Good catch, done.